internal/framework/resources/registered_domain/update.go

```diff
@@ -0,0 +1,236 @@
+package registered_domain
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/dnsimple/dnsimple-go/dnsimple"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/terraform-providers/terraform-provider-dnsimple/internal/consts"
+	"github.com/terraform-providers/terraform-provider-dnsimple/internal/framework/utils"
+)
+
+func (r *RegisteredDomainResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	var (
+		configData *RegisteredDomainResourceModel
+		planData   *RegisteredDomainResourceModel
+		stateData  *RegisteredDomainResourceModel
+	)
+
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &planData)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	resp.Diagnostics.Append(req.State.Get(ctx, &stateData)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	resp.Diagnostics.Append(req.Config.Get(ctx, &configData)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	if planData.ContactId.ValueInt64() != stateData.ContactId.ValueInt64() {
+		resp.Diagnostics.AddError(
+			fmt.Sprintf("contact_id change not supported: %s, %d", planData.Name.ValueString(), planData.Id.ValueInt64()),
+			"contact_id change not supported by the DNSimple API",
+		)
+		return
+	}
+
+	if !planData.ExtendedAttributes.Equal(stateData.ExtendedAttributes) {
+		resp.Diagnostics.AddError(
+			fmt.Sprintf("extended_attributes change not supported: %s, %d", planData.Name.ValueString(), planData.Id.ValueInt64()),
+			"extended_attributes change not supported by the DNSimple API",
+		)
+		return
+	}
+
+	domainRegistration, diags := getDomainRegistration(ctx, stateData)
+	if diags.HasError() {
+		resp.Diagnostics.Append(diags...)
+		return
+	}
+
+	if domainRegistration.Id.IsNull() {
+		resp.Diagnostics.AddError(
+			"failed to read domain registration from state",
+			"domain registration is nil. This should not happen. Please remove the resource from the state and try importing.",
+		)
+		return
+	}
+
+	lowerCaseDomainName := strings.ToLower(configData.Name.ValueString())
+	response, err := r.config.Client.Registrar.GetDomainRegistration(ctx, r.config.AccountID, lowerCaseDomainName, strconv.Itoa(int(domainRegistration.Id.ValueInt64())))
+
+	if err != nil {
+		var errorResponse *dnsimple.ErrorResponse
+		if errors.As(err, &errorResponse) {
+			resp.Diagnostics.Append(utils.AttributeErrorsToDiagnostics(errorResponse)...)
+			return
+		}
+
+		resp.Diagnostics.AddError(
+			"failed to register DNSimple Domain",
+			err.Error(),
+		)
+		return
+	}
+
+	// Check if the domain is in registered state
+	if response.Data.State != consts.DomainStateRegistered {
+		timeouts, diags := getTimeouts(ctx, planData)
+		resp.Diagnostics.Append(diags...)
+		if resp.Diagnostics.HasError() {
+			return
+		}
+
+		err := utils.RetryWithTimeout(ctx, func() (error, bool) {
```

Comment:
I would suggest abstracting this as it's identical to the code in `create.go`.

Reply:
Done